mypy/messages.py

```diff
@@ -522,7 +522,13 @@ def incompatible_argument(self, n: int, m: int, callee: CallableType, arg_type:
             name = callee.name[1:-1]
             n -= 1
             msg = '{} item {} has incompatible type {}'.format(
-                name[0].upper() + name[1:], n, self.format_simple(arg_type))
+                name.title(), n, self.format_simple(arg_type))
+        elif callee.name == '<dict>':
+            name = callee.name[1:-1]
+            n -= 1
+            key_type, value_type = cast(TupleType, arg_type).items
+            msg = '{} entry {} has incompatible type {}: {}'.format(
```

Comment:
I've seen "entry" used for dictionaries (example: https://docs.python.org/3/reference/expressions.html#dictionary-displays) quite often; item too but it's more ambiguous (it's also used for lists). I don't mind changing it to avoid bikeshedding if you still prefer it.